cmd/snap-bootstrap/bootstrap/bootstrap.go

```diff
@@ -92,6 +93,15 @@ func Run(gadgetRoot, device string, options Options) error {
 	if err := diskLayout.RemoveCreated(); err != nil {
 		return fmt.Errorf("cannot remove partitions from previous install: %v", err)
 	}
+	// at this point we removed any existing partition, nuke any
+	// of the existing sealed key files placed outside of the
+	// encrypted partitions (LP: #1879338)
+	if options.KeyFile != "" {
```

Comment:
just an FYI @cmatsuoka already add this code to https://github.com/snapcore/snapd/pull/8591 but in a different place and we had a bit of discussion around where this code should live. I thought it should live here but Claudio pointed out a good point that at this point we have not yet attempted to clear the TPM, so if we are in this codepath by accident we could be inadvertently making data irreversibly unavailable, whereas with the place Claudio put it, we have already re-provisioned the TPM so the keyfile is guaranteed to be useless when we delete it there.

Reply:
I'm actually still not fully understanding the reasoning there, independently of the state of the tpm, we just removed the partitions so recovering is probably not impossible but hard

Reply:
yes I agree that at this point we've already deleted the partitions and re-created them in presumably the same location, so at least some of the content of the original partitions may have been overwritten at this point making recovering quite difficult at least. Anyways I just wanted to connect the dots here, perhaps when @cmatsuoka he can comment as well if there's additional reasons to not put it here and put it there instead

Reply:
At that point the partitions are gone so it doesn't matter. Looking at the whole story I agree that this is the best approach so we reached consensus here. Once this lands I'll adjust #8591 accordingly.

(Regarding TPM I think we should probe if it's already provisioned at the start of the installation process, without attempting to re-provision it, to avoid installing the system and then fail at the end.)